### PR TITLE
Implement a 'dynamic_groups' field on the DeviceType for GraphQL.

### PR DESCRIPTION
**Pull request by Kircheneer**
# Closes: #2307

# What's Changed

- Implement a `dynamic_groups` field on the GraphQL `DeviceType` that shows to which dynamic groups the device belongs to
- Statically create `DynamicGroupType` because otherwise I would get the following error

```
Exception Value: 	

Could not import 'nautobot.core.graphql.schema_init.schema' for Graphene setting 'SCHEMA'. ImportError: Module "nautobot.extras.graphql.types" does not define a "DynamicGroupType" attribute/class.
```

Example screenshot:
Screenshot 2022-08-26 at 09 28 30

# TODO
- [x] Explanation of Change(s)
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by bryanculver — COMMENTED** [review]
😎 

Can we see if we can get all of the existing DynamicGroup supporting models (there aren't many at this time) to support this?

As well can we get at least a test case for Device to ensure we get the results we expect?

**Comment by Kircheneer** [comment]
Expanded the availability of the key to all relevant models and added a test. Let me know if you want more or more thorough testing.

File: nautobot/dcim/graphql/types.py [hunk 1]
```diff
@@ -53,6 +54,11 @@ class Meta:
         filterset_class = DeviceFilterSet
         exclude = ["_name"]
 
+    dynamic_groups = graphene.List("nautobot.extras.graphql.types.DynamicGroupType")
+
+    def resolve_dynamic_groups(self, args):
+        return DynamicGroup.objects.get_for_object(Device.objects.get(pk=self.pk))
```
Comment (bryanculver) [inline nautobot/dcim/graphql/types.py:60]: Until we refactor this as a Mixin or base, any reason why to not change to in all places?

```suggestion
        return DynamicGroup.objects.get_for_object(self.Meta.model.objects.get(pk=self.pk))
```

Just to minimize copy-paste issues.

Comment (glennmatthews) [reply nautobot/dcim/graphql/types.py:60]: Is this level of indirection needed or does `DynamicGroup.objects.get_for_object(self)` work too?

Comment (bryanculver) [reply nautobot/dcim/graphql/types.py:60]: > Is this level of indirection needed or does `DynamicGroup.objects.get_for_object(self)` work too?

Is `self` a `Device` object? or a shadowed one for GraphQL? I'm not familiar how the GraphQL types are built in relation to the normal models.

Comment (Kircheneer) [reply nautobot/dcim/graphql/types.py]: It is not in fact needed, it works with `DynamicGroup.objects.get_for_object(self)`.

**Review by glennmatthews — APPROVED** [review]

**Review by bryanculver — APPROVED** [review]

**Pull request merged**